stdlib/universe/window2.go

```diff
@@ -0,0 +1,490 @@
+package universe
+
+import (
+	"context"
+	"sort"
+
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/memory"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/arrowutil"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/internal/execute/dataset"
+	"github.com/influxdata/flux/internal/execute/table"
+	"github.com/influxdata/flux/internal/mutable"
+	"github.com/influxdata/flux/interval"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/values"
+)
+
+func init() {
+	plan.RegisterPhysicalRules(WindowSortedInputRule{})
```

Comment:
Is there a mechanism in place so that this doesn't happen before a group pushdown rewrite and therefore stop it from matching?

Reply:
There isn't really. I think the cost mechanic is supposed to choose one of the paths, but we've never really used cost. This shouldn't stop a group pushdown from happening though because it will resolve in one of two ways:

1. `ReadRange |> group |> window` -> `ReadGroup |> window`
2. `ReadRange |> group |> window` -> `ReadRange |> group |> sort |> window` -> `ReadGroup -> sort -> window`

We can add another planner rule into influxdb specifically to resolve the other side of this to either always add the sort or to remove the sort if it's not needed. I think it might be better even with the push down but it really depends on if `ReadGroup` returns the data sorted by time or not.